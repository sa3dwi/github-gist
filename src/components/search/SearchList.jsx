import React from 'react';

import SearchItem from './SearchItem';

const SearchList = ({searchResults, errorMessage}) => {

    return (
        <div className="wrap" data-testid="searchListWarp">
            <div className="container">
            {!errorMessage && searchResults.length > 0 ? 
                <ul className="flex-list" data-testid="searchList">
                {searchResults.map((item, index) => {
                    return (
                    <div key={index}>
                        <SearchItem item={item}/> 
                        <br/><hr/>
                    </div>
                    );
                })}
                </ul> 
                : 
                <div className="row d-flex justify-content-center alert">
                {errorMessage}
                </div> 
            }
            </div>
      </div>
    );
    
};

export default SearchList;