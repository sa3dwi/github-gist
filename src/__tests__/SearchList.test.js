import React from 'react';
import { render, cleanup } from '@testing-library/react'
import { create } from "react-test-renderer";
import { expect } from 'chai';

import SearchList from './../components/search/SearchList';

beforeEach(() => {
  localStorage.clear();
  cleanup
})

describe("SearchList component UI", () => {
  test('matches the snapshot', () => {
    const wrapper = create(<SearchList searchResults={[]} />)
    expect(wrapper.toJSON()).to.matchSnapshot()
  })
})

// TODO: use mock api 
describe("SearchList component Functionality", () => {
  test('allows you show search list', () => {
    render(<SearchList searchResults={[]} />)
  })
})
