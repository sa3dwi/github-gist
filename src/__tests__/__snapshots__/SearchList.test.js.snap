// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchList component UI matches the snapshot 1`] = `
<div
  className="wrap"
  data-testid="searchListWarp"
>
  <div
    className="container"
  >
    <div
      className="row d-flex justify-content-center alert"
    />
  </div>
</div>
`;
